File: InMotionGIT.Common/Helpers/KeyValidator.vb
Namespace Helpers

    Public NotInheritable Class KeyValidator

        ''' <summary>
        ''' It generates a key based on the number of the week in the month the number of concatenated day of the week.
        ''' </summary>
        ''' <returns>Key generated</returns>
        Public Shared Function GenerateKey() As String
            Return String.Format("{0}{1}", Date.Now.WeekOfMonth, Date.Now.NumericDayOfWeek)
        End Function

        ''' <summary>
        ''' Validator key for boor
        ''' </summary>
        ''' <param name="key"></param>
        ''' <returns></returns>
        Public Shared Function KeyValidator(key As String) As Boolean
            Dim result As String = GenerateKey()
            Return result.Equals(key)
        End Function

    End Class

End Namespace

File: Common/InMotionGIT.Common/Helpers/KeyValidator.cs
using System;
using InMotionGIT.Common.Extensions;

namespace InMotionGIT.Common.Helpers
{

    public sealed class KeyValidatorType
    {

        /// <summary>
        /// It generates a key based on the number of the week in the month the number of concatenated day of the week.
        /// </summary>
        /// <returns>Key generated</returns>
        public static string GenerateKey()
        {
            return string.Format("{0}{1}", DateTime.Now.WeekOfMonth(), DateTime.Now.NumericDayOfWeek());
        }

        /// <summary>
        /// Validator key for boor
        /// </summary>
        /// <param name="key"></param>
        /// <returns></returns>
        public static bool KeyValidator(string key)
        {
            string result = GenerateKey();
            return result.Equals(key);
        }

    }

}